File: python/codingBat/String-2/double_char.py
#!/usr/bin/python3
# Given a string, return a string where for every char in the original, there are two chars.

def double_char(str):
  newStr = ""
  for i in range(len(str)):
    newStr = newStr + str[i:i+1] + str[i:i+1]
  return newStr

print(double_char('Hi-There'))
print(double_char('Word!'))
print(double_char('aa'))

File: java/codingBat/String-2/doubleChar.java
//Given a string, return a string where for every char in the original, there are two chars.
//doubleChar("The") → "TThhee"
//doubleChar("AAbb") → "AAAAbbbb"
//doubleChar("Hi-There") → "HHii--TThheerree"

public class doubleChar{
    public static void main(String... args) {
        System.out.println(doubleChar("The"));
        System.out.println(doubleChar("AAbb"));
        System.out.println(doubleChar("Hi-There"));

    }
    public static String doubleChar(String str) {
        String subString = "";
        int l = str.length();
        for (int x = 0; x < l; x++) {
            subString = subString + str.substring(x, x + 1) + str.substring(x, x + 1);
        }
        return subString;
    }
}